src/main/java/eu/donals/Player.java

```diff
@@ -0,0 +1,88 @@
+package eu.donals;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Player {
+    private String name;
+    private List<String> inventory;
+    private UserInterface ui;
+
+    public boolean hasItem(String item) { return inventory.contains(item); }
+
+    public String getPlayerName() { return name; }
+
+    public Location move(String argument, GameState state) throws IOException {
+        String[] currNeighbours = state.getLocation().getNeighbours();
+        Loader locationLoader = new Loader();
+        Location location = null;
+        List<String> outputText = new ArrayList<String>();
+
+        for (String currNeighbour: currNeighbours) {
+            if (currNeighbour.equals(argument)) {
+                location =  locationLoader.parseLocationByName(argument, state.getLocationList());
+
+                outputText.add("eu.donals.Location: " + location.getName());
+                outputText.add(location.getDescription());
+                outputText.add("NPCs: " + location.getNpc());
+                outputText.add("Locations near by: " + Arrays.toString(location.getNeighbours()));
+
+                ui.displayText(outputText);
+
+                return location;
+            }
+        }
+
+        ui.displayText("Cannot find that place");
+        return state.getLocation();
+    }
+
+    public void look(String argument, GameState state) throws IOException {
+        if(argument.equals(state.getLocation().getNpc())) {
+            ui.displayText(state.getLocation().getNpcAbout());
+        } else {
+            ui.displayText("Npc is not here");
```

Comment:
Updating the knowledge of the user through messages could be done slightly more often, not only when an error occurs.